ni_measurementlink_service/session_management/_reservation.py

```diff
@@ -94,6 +142,52 @@ def _grpc_channel_pool(self) -> GrpcChannelPool:
             raise ValueError("This method requires a gRPC channel pool.")
         return self._session_manager._grpc_channel_pool
 
+    @cached_property
```

Comment:
I'm really jealous of this decorator. It allows you to write the code in an intuitive way and have python make it efficient for you. LabVIEW, on the other hand......